rust/lancedb/src/remote/db.rs

```diff
@@ -243,13 +252,37 @@ impl<S: HttpSend> ConnectionInternal for RemoteDatabase<S> {
     }
 }
 
+/// RemoteOptions contains a subset of StorageOptions that are compatible with Remote LanceDB connections
+#[derive(Clone, Debug, Default)]
+pub struct RemoteOptions(pub HashMap<String, String>);
+
+impl RemoteOptions {
+    pub fn new(options: HashMap<String, String>) -> Self {
+        Self(options)
+    }
+}
+
+impl From<StorageOptions> for RemoteOptions {
```

Comment:
Should we consider using TryFrom here in case there's a passed storage option that can't be converted to a remote option, we could return an error rather than ignoring it? WDYT?

Reply:
I don't think so, because we explicitly allow specific storage options as remote options, all others would be ignored. Unless we want to add additional validation for values that are remote-specific. I don't think we would need that.